processor/servicegraphprocessor/factory_test.go

```diff
@@ -49,20 +49,29 @@ func TestNewProcessor(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			// Prepare
 			factory := NewFactory()
+			connectorFactory := NewConnectorFactory()
```

Comment:
nitpick: Maybe it is worth adding a separate test case or the test.

Reply:
I was on the fence about this one. I had to pick and choose what to do in parallel vs fully copy. I was on the fence about this one.

Reply:
Done